lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/TableIndexDAOImpl.java

```diff
@@ -960,17 +961,15 @@ public long getSumOfFileSizes(List<Long> rowIds) {
 	}
 
 	@Override
-	public List<SynapseStorageProjectStats> getSynapseStorageStats() {
-		return template.query(
-				SQL_SELECT_PROJECTS_BY_SIZE,
-				(rs, rowNum) -> {
-					SynapseStorageProjectStats stats = new SynapseStorageProjectStats();
-					stats.setId(rs.getString(1));
-					stats.setProjectName(rs.getString(2));
-					stats.setSizeInBytes(rs.getLong(3));
-					return stats;
-				});
-
+	public void streamSynapseStorageStats(Callback<SynapseStorageProjectStats> callback) {

```

Comment:
Stream one row at a time to the callback, so we don't load the entire call into memory